runtime/reference/lsp.md

```diff
@@ -1,14 +1,224 @@
 ---
-title: "Testing API"
+title: "The Language Server"
+oldUrl:
+- /runtime/manual/advanced/language_server/overview/
+- /runtime/manual/advanced/language_server/imports/
+- /runtime/manual/advanced/language_server/testing_api/
 ---
 
+The Deno CLI comes with a built in language server that can provide an
+intelligent editing experience, as well as a way to easily access the other
+tools that come built in with Deno. For most users, using the language server
+would be via an editor such as
+[Visual Studio Code](/runtime/manual/references/vscode_deno/) or
+[other editors](/runtime/getting_started/setup_your_environment/).
+
+This page is designed for those creating integrations to the language server or
+providing a package registry for Deno that integrates intelligently.
+
+The Deno Language Server provides a server implementation of the
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+which is specifically tailored to provide a _Deno_ view of code. It is
+integrated into the command line and can be started via the `lsp` sub-command.
+
+## Structure
+
+When the language server is started, a `LanguageServer` instance is created
+which holds all of the state of the language server. It also defines all of the
+methods that the client calls via the Language Server RPC protocol.
+
+## Settings
+
+The language server supports a number of settings for a workspace:
+
+- `deno.enable`
+- `deno.enablePaths`
+- `deno.cache`
+- `deno.certificateStores`
+- `deno.config`
+- `deno.importMap`
+- `deno.internalDebug`
+- `deno.codeLens.implementations`
+- `deno.codeLens.references`
+- `deno.codeLens.referencesAllFunctions`
+- `deno.codeLens.test`
+- `deno.suggest.completeFunctionCalls`
+- `deno.suggest.names`
+- `deno.suggest.paths`
+- `deno.suggest.autoImports`
+- `deno.suggest.imports.autoDiscover`
+- `deno.suggest.imports.hosts`
+- `deno.lint`
+- `deno.tlsCertificate`
+- `deno.unsafelyIgnoreCertificateErrors`
+- `deno.unstable`
+
+And there are settings that are supported on a per resource basis by the
+language server:
+
+- `deno.enable`
+- `deno.enablePaths`
+- `deno.codeLens.test`
+
+Deno analyzes these settings at multiple points in the language server process.
+First, when the `initialize` request from the client, the
+`initializationOptions` will be assumed to be an object that represents the
+`deno` namespace of options. For example, the following value will enable Deno
```

Comment:
Something is off with that sentence. It feels like a verb or something is missing in the `initialize` part